File: source-connector/src/main/java/sourceconnector/service/offset/OffsetRecordRepository.java
package sourceconnector.service.offset;

import offsetmanager.domain.file.FileKey;
import offsetmanager.domain.offset.OffsetRecord;

import java.util.List;
import java.util.Optional;

public interface OffsetRecordRepository extends AutoCloseable{
  Optional<OffsetRecord> findLastOffsetRecord(FileKey key);
  List<OffsetRecord> findLastOffsetRecords(List<FileKey> keys);
}

File: source-connector/src/main/kotlin/sourceconnector/service/offset/OffsetRecordRepository.kt
package sourceconnector.service.offset

import offsetmanager.domain.file.FileKey
import offsetmanager.domain.offset.OffsetRecord
import java.lang.AutoCloseable
import java.util.*

interface OffsetRecordRepository : AutoCloseable {
  fun findLastOffsetRecord(key: FileKey): OffsetRecord?
  fun findLastOffsetRecords(keys: List<FileKey>): List<OffsetRecord>
}
